docs/reference/search/retriever.asciidoc

```diff
@@ -257,27 +371,117 @@ Sets a minimum threshold score for including documents in the re-ranked results.
 
 A text similarity re-ranker retriever is a compound retriever. Child retrievers may not use elements that are restricted by having a compound retriever as part of the retriever tree.
 
-===== Example
+[discrete]
+[[text-similarity-reranker-retriever-example-cohere]]
+==== Example: Cohere Rerank
+
+This example enables out-of-the-box semantic search by reranking top documents using the Cohere Rerank API. This approach eliminate the need to generate and store embeddings for all indexed documents.
+This requires a <<infer-service-cohere,Cohere Rerank inference endpoint>> using the `rerank` task type.
 
 [source,js]
 ----
 GET /index/_search
 {
-    "retriever": {
-        "text_similarity_reranker": {
-            "retriever": {
-                "standard": { ... }
-            },
-            "field": "text",
-            "inference_id": "my-cohere-rerank-model",
-            "inference_text": "Most famous landmark in Paris",
-            "rank_window_size": 100,
-            "min_score": 0.5
+   "retriever":{
+      "text_similarity_reranker":{
+         "retriever":{
+            "standard":{
+               "query":{
+                  "match_phrase":{
+                     "text":"landmark in Paris"
+                  }
+               }
+            }
+         },
+         "field":"text",
+         "inference_id":"my-cohere-rerank-model",
+         "inference_text":"Most famous landmark in Paris",
+         "rank_window_size":100,
+         "min_score":0.5
+      }
+   }
+}
+----
+// NOTCONSOLE
+
+[discrete]
+[[text-similarity-reranker-retriever-example-eland]]
+==== Example: Semantic reranking with a Hugging Face model
+
+The following example uses the `cross-encoder/ms-marco-MiniLM-L-6-v2` model from Hugging Face to rerank search results based on semantic similarity.
+The model must be uploaded to {es} using https://www.elastic.co/guide/en/elasticsearch/client/eland/current/machine-learning.html#ml-nlp-pytorch[Eland].
+
+[TIP]
+====
+Refer to {ml-docs}/ml-nlp-model-ref.html#ml-nlp-model-ref-text-similarity[the Elastic NLP model reference] for a list of third party text similarity models supported by {es}.
+====
+
+Follow these steps to load the model and create a semantic reranker.
+
+. Install Eland using `pip`
++
+[source,sh]
+----
+python -m pip install eland[pytorch]
+----
++
+. Upload the model to {es} using Eland. This example assumes you have an Elastic Cloud deployment and an API key. Refer to the https://www.elastic.co/guide/en/elasticsearch/client/eland/current/machine-learning.html#ml-nlp-pytorch-auth[Eland documentation] for more authentication options.
++
+[source,sh]
+----
+eland_import_hub_model \
+  --cloud-id $CLOUD_ID \
+  --es-api-key $ES_API_KEY \
+  --hub-model-id cross-encoder/ms-marco-MiniLM-L-6-v2 \
+  --task-type c \
```

Comment:
Does this work like this? I think it has to be an exact value from an enum, in this case `text_similarity`.

Reply:
that's a fat-fingered typo :)